src/cli.ts

```diff
@@ -1,30 +1,25 @@
 #!/usr/bin/env node
+import { defineCommand, runMain } from 'citty'
+import type { CommandDef } from 'citty'
 
-/* eslint-disable no-console */
-import mri from 'mri'
-import { resolve } from 'pathe'
-import { buildModule } from './build'
-import { prepareModule } from './prepare'
+const _rDefault = (r: any) => (r.default || r) as Promise<CommandDef>
 
-// TODO: use citty
-function main () {
-  const args = mri(process.argv.slice(2))
-
-  if (args._[0] === 'prepare') {
-    return prepareModule({
-      rootDir: resolve(args._[1] || '.')
-    })
+const main = defineCommand({
+  meta: {
+    name: 'nuxt-module-build',
+    description: 'Nuxt Module Builder'
+  },
+  subCommands: {
+    prepare: () => import('./commands/prepare').then(_rDefault),
+    build: () => import('./commands/build').then(_rDefault)
+  },
+  setup (context) {
+    // TODO: support 'default command' in citty?
```

Comment:
Actually setup is _meant_ to be default command (in citty next major, it won't continue with sub commands) however i think we might start deprecating implicit sub-commands like this across CLIs just like we did for `nuxt <dev>` in nuxt 3